### PR TITLE
Add challenge form new

### PR DESCRIPTION
**Pull request by salaos**

### CODE REVIEW

File: components/Challenges.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { render } from "@testing-library/react";
+
+import { challenges } from "../pages/api/db";
+
+// test("renders a card", () => {
```
Comment (augustinanders) [inline components/Challenges.test.js:6]: Either you implement your Tests, or you can get rid of them.

File: components/ChallengeForm.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  StyledBox,
+  StyledRange,
+  StyledButton,
+} from "../components/StyledComponents";
+
+// create a callback function that is called when the form is submitted
+// the callback function should create a new challenge object
+// and call the onAddChallenge function that is passed as a prop
+
+async function persistAddedChallenge(challenge) {
```
Comment (augustinanders) [inline components/ChallengeForm.js:16]: I don't understand whats every Detail here, but the preview(codesandbox) throws an Runtime-Error, when trying to fetch.

I do not really understand why you are using an API, and fetchting/posting at this point. 

Couldn't you just use a State to Store your Data persistent? (LocalStorage)

**Review by augustinanders — CHANGES_REQUESTED** [review]
Hey Jana, Overall your App looks very good to me!🚀

Since the PR was very extensive I couldn't comment on everything. Left you a few comments below. Somehow the fetching is not working in the Sandbox preview. 

Also make sure to host your project on Vercel to make QA easier later. 

For the future I would recommend to go smaller steps. This way you make it easier to review your code, and it will speed up the production.

File: components/ChallengeForm.js (hunk 2 above)
Comment (salaos) [reply components/ChallengeForm.js:16]: Ah thanks for the error info!
You're right, I guess that would be much simpler. Correct me if I'm wrong but I thought I need to use an api to use the CRUD operations more easily to change & delete the data in the next step

**Pull request merged**